streams/src/test/java/org/apache/kafka/streams/processor/internals/InternalTopologyBuilderTest.java

```diff
@@ -175,6 +177,12 @@ public void testAddProcessorWithNullParents() {
         builder.addProcessor("processor", new MockProcessorSupplier<>(), (String) null);
     }
 
+    @Test(expected = TopologyException.class)
+    public void testAddProcessorWithBadSupplier() {
+        final Processor<Object, Object> processor = new MockProcessor<>();
+        builder.addProcessor("processor", () -> processor, (String) null);
```

Comment:
We should use `aasertThrows` and verify the error message similar to the `TransformerSupplier` test